bokehjs/gulp/eco.coffee

```diff
@@ -1,14 +1,59 @@
 path = require 'path'
-merge = require 'merge'
 through = require 'through2'
 gutil = require 'gulp-util'
 coffee = require 'coffee-script'
 eco = require 'eco'
 
-#{preprocess} = require "eco/preprocessor"
+{preprocess} = require "eco/src/preprocessor"
+{indent}     = require "eco/src/util"
 
-#precompile = (source) ->
-#  script = coffee.compile(preprocess(source), {noWrap: true})
+precompile = (source) ->
+  script = coffee.compile(preprocess(source), {noWrap: true})
+
+  return """
+    function(__obj) {
+      if (!__obj) __obj = {};
+      var __out = [];
+      var __capture = function(callback) {
+        var out = __out, result;
+        __out = [];
+        callback.call(this);
+        result = __out.join('');
+        __out = out;
+        return __safe(result);
+      };
+      var __sanitize = function(value) {
+        if (value && value.ecoSafe) {
+          return value;
+        } else if (typeof value !== 'undefined' && value != null) {
+          return __escape(value);
+        } else {
+          return '';
+        }
+      };
+      var __safe = function(value) {
+        if (value && value.ecoSafe) {
+          return value;
+        } else {
+          if (!(typeof value !== 'undefined' && value != null)) value = '';
+          var result = new String(value);
+          result.ecoSafe = true;
+          return result;
+        }
+      };
+      var __escape = function(value) {
+        return ('' + value)
+          .replace(/&/g, '&amp;')
+          .replace(/</g, '&lt;')
+          .replace(/>/g, '&gt;')
+          .replace(/\x22/g, '&quot;');
+      };
+      (function() {
+    #{indent(script, 4)}
+      }).call(__obj);
+      return __out.join('');
+    }
```

Comment:
Previously this code modified `_obj` with `escape` and `safe` function. So, if you passed `@model.attributes` to a template, you got two new model attributes for free.